src/check_release.rs

```diff
@@ -249,66 +250,77 @@ pub(super) fn run_check_release(
         .collect::<anyhow::Result<Vec<_>>>()?;
 
     let mut results_with_errors = vec![];
+    let mut results_with_warnings = vec![];
     for (semver_query, time_to_decide, results) in all_results {
         config
             .log_verbose(|config| {
                 let category = match overrides.effective_required_update(semver_query) {
                     RequiredSemverUpdate::Major => "major",
                     RequiredSemverUpdate::Minor => "minor",
                 };
-                if results.is_empty() {
-                    writeln!(
-                        config.stderr(),
-                        "{}{:>12}{} [{:8.3}s] {:^18} {}",
-                        Style::new()
-                            .fg_color(Some(Color::Ansi(AnsiColor::Green)))
-                            .bold(),
-                        "PASS",
-                        Reset,
-                        time_to_decide.as_secs_f32(),
-                        category,
-                        semver_query.id
-                    )?;
-                } else {
-                    writeln!(
-                        config.stderr(),
-                        "{}{:>12}{} [{:>8.3}s] {:^18} {}",
-                        Style::new()
-                            .fg_color(Some(Color::Ansi(AnsiColor::Red)))
-                            .bold(),
-                        "FAIL",
-                        Reset,
-                        time_to_decide.as_secs_f32(),
-                        category,
-                        semver_query.id
-                    )?;
-                }
+
+                let (status, status_color) = match (
+                    results.is_empty(),
+                    overrides.effective_lint_level(semver_query),
+                ) {
+                    (true, _) => ("PASS", AnsiColor::Green),
+                    (false, LintLevel::Deny) => ("FAIL", AnsiColor::Red),
+                    (false, LintLevel::Warn) => ("WARN", AnsiColor::Yellow),
+                    (false, LintLevel::Allow) => unreachable!(),
+                };
+
+                writeln!(
+                    config.stderr(),
+                    "{}{:>12}{} [{:8.3}s] {:^18} {}",
+                    Style::new()
+                        .fg_color(Some(Color::Ansi(status_color)))
+                        .bold(),
+                    status,
+                    Reset,
+                    time_to_decide.as_secs_f32(),
+                    category,
+                    semver_query.id
+                )?;
                 Ok(())
             })
             .expect("print failed");
+
         if !results.is_empty() {
-            results_with_errors.push((semver_query, results));
+            match overrides.effective_lint_level(semver_query) {
+                LintLevel::Deny => results_with_errors.push((semver_query, results)),
+                LintLevel::Warn => results_with_warnings.push((semver_query, results)),
+                _ => (),
```

Comment:
This is probably unreachable too, right?

Am I correct in thinking that the invariant is "`LintLevel::Allow` lints are skipped entirely, because there's no point in running them"?